pkg/apis/pipeline/v1beta1/param_types.go

```diff
@@ -107,17 +132,23 @@ type ArrayOrString struct {
 	Type      ParamType `json:"type"` // Represents the stored type of ArrayOrString.
 	StringVal string    `json:"stringVal"`
 	// +listType=atomic
-	ArrayVal []string `json:"arrayVal"`
+	ArrayVal  []string          `json:"arrayVal"`
+	ObjectVal map[string]string `json:"objectVal"`
```

Comment:
Originally nested objects wouldn't be supported. Changing it to map[string]ArrayOrString can be a bit tricky for testing and capturing all the corner cases. This would be good stretches to be done after the feature is launched with alpha guard flags